test/functional/page_objects/dashboard_page.js

```diff
@@ -220,12 +220,14 @@ export function DashboardPageProvider({ getService, getPageObjects }) {
 
     async clickNewDashboard() {
       // newDashboardLink button is only visible when dashboard listing table is displayed (at least one dashboard).
-      const exists = await testSubjects.exists('newDashboardLink');
+      let exists = await testSubjects.exists('newItemButton');
       if (exists) {
-        return await testSubjects.click('newDashboardLink');
+        return await testSubjects.click('newItemButton');
       }
 
-      // no dashboards exist, click createDashboardPromptButton to create new dashboard
+      exists = await testSubjects.exists('createDashboardPromptButton');
```

Comment:
Is an assertion whether exists is true missing here?

Reply:
There is a separate test to check whether the buttons appear at the correct time ('create prompt' test).
This function assumes that the buttons appear correctly, so the second await assignment is not necessary. Also added a comment to the function.